src/app-layer-htp.c

```diff
@@ -1627,8 +1627,12 @@ static int HtpResponseBodyHandle(HtpState *hstate, HtpTxUserData *htud,
 
     int result = 0;
 
-    /* see if we need to open the file */
-    if (!(htud->tcflags & HTP_FILENAME_SET))
+    /* see if we need to open the file
```

Comment:
I don't understand this fix. This function is being called by HTPCallbackResponseBodyData, so the response body callback. The patch seems to suggest we can get called while there is actually no response body data. Why are we called then? And shouldn't the fix be that the callback is not called?

Reply:
`HTPCallbackResponseBodyData` is called with data

The patch is about handling junk data before the actual response line (and dealing with HTTP 0.9 as well).

Should this junk data be thrown out ?

Reply:
I think this junk data should then be stripped by libhtp before invoking the callback.

Reply:
I would like to be sure that I understand "be stripped by" : 
Do you mean that Suricata should not send these bytes to libhtp callback ?

I did not choose this option because of another case : if we have more than one request and reply in the same connection.
Then, starting from the second response, it is impossible to tell if this is junk data before the response line, or extra data (more than content-length) that will be interpreted by browsers (there are http-evader cases like this).
So, is it right to do differently for the first reply than for the next ones ?